### ISSUE TITLE
Location Icon Does Not Move Unless Map Is Interacted With

### ISSUE THREAD

**Issue by brswan** [Mapsui/Mapsui#708]
Platform: Android 8.0
Mapsui: 2.0.0-beta.26
Xam.Plugin.Geolocator: v4.6.1-beta
Xamarin Forms: 4.0.0.425677

I am following the sample code and using the same methods to update my location with the current gps coordinate.  The Location marker does not move to the location after calling MyMapView.MyLocationLayer.UpdateMyLocation.  I have to interact with the map in order for the location marker to move.

```
 public MainPage()
        {
            InitializeComponent();

            if (m_UseOfflineMaps)
            {
                string documentDirectory = DependencyService.Get<IDocumentDirectoryProvider>().GetDocumentsDirectory();
                string tilesPath = Path.Combine(documentDirectory, "world.mbtiles");

                if (File.Exists(tilesPath))
                {
                    TileLayer mbTileLayer = CreateMbTilesLayer(tilesPath);
                    MyMapView.Map.Layers.Add(mbTileLayer);
                }
            }

            StartGPS();
        }

private async void StartGPS()
        {
            await CrossGeolocator.Current.StartListeningAsync(TimeSpan.FromSeconds(1),
                1,
                true,
                new ListenerSettings
                {
                    ActivityType = ActivityType.Fitness,
                    AllowBackgroundUpdates = false,
                    DeferLocationUpdates = true,
                    DeferralDistanceMeters = 1,
                    DeferralTime = TimeSpan.FromSeconds(0.2),
                    ListenForSignificantChanges = false,
                    PauseLocationUpdatesAutomatically = true
                });

            CrossGeolocator.Current.PositionChanged += MyLocationPositionChanged;
            //CrossGeolocator.Current.PositionError += MyLocationPositionError;
        }

        private void MyLocationPositionChanged(object sender, PositionEventArgs e)
        {
            Device.BeginInvokeOnMainThread(() =>
            {
                MyMapView.MyLocationLayer.UpdateMyLocation(new Mapsui.UI.Forms.Position(e.Position.Latitude, e.Position.Longitude));
                MyMapView.MyLocationLayer.UpdateMyDirection(e.Position.Heading, MyMapView.Viewport.Rotation);
            });
        }
```

**Comment by pauldendulk** [comment]
Seems like something that should work that way.

[Here is the code](https://github.com/Mapsui/Mapsui/blob/master/Mapsui.UI.Forms/Objects/MyLocationLayer.cs#L147) of UpdateMyLocation. MapControl.RefreshGraphics is called. I did not try to reproduce this.

**Comment by brswan** [comment]
Do you want me to include a sample project with the issue?

**Comment by pauldendulk** [comment]
@charlenni Could it be that this was fixed with your recent changes?

**Comment by charlenni** [comment]
I‘m not sure if this is solved. I had to test it.

**Comment by pauldendulk** [comment]
Okay, I will reopen and add the needs-testing tag.

**Comment by pauldendulk** [comment]
We assume this is fixed. Please let us know when it is not

**Comment by taggreenleaf** [comment]
Cloning the repository yesterday and running the samples, I see the same behavior. Own location is at 0,0. I  drag the map slightly and then the marker slides towards my real location and after two seconds, it lands where it should.
When I then enter another of the samples, the marker is at 0,0 and interacting with the map does not make it move. Going to the previous sample, where I made it update the marker, the marker is at 0,0 and interacting will not no longer make it update.

I also tried making an empty project (with location permissions) using v2.0.3 and here the marker is at 0,0 but even interaction will not make it move, so it does not seem like it is fixed
